RealmSwift/Tests/MixedCollectionTest.swift

```diff
@@ -0,0 +1,990 @@
+        XCTAssertTrue(mixedObject.anyCol == mixedObject2.anyCol, "instances should be equal by `==` operator")
+        XCTAssertTrue(mixedObject.isEqual(mixedObject2), "instances should be equal by `isEqual` method")
+        XCTAssertEqual(mixedObject.anyCol.dictionaryValue, mixedObject.anyCol.dictionaryValue)
+
+        let realm = realmWithTestPath()
+        try realm.write {
+            realm.add(mixedObject)
+            realm.add(mixedObject2)
+        }
+
+        XCTAssertEqual(mixedObject.anyCol, mixedObject2.anyCol)
+        XCTAssertTrue(mixedObject.anyCol == mixedObject2.anyCol, "instances should be equal by `==` operator")
+        XCTAssertTrue(mixedObject.isEqual(mixedObject2), "instances should be equal by `isEqual` method")
+
+        let mixedObject3 = ModernAllTypesObject()
+        mixedObject3.anyCol = AnyRealmValue.fromDictionary(dictionary)
+
+        try realm.write {
+            realm.add(mixedObject3)
+        }
+
+        XCTAssertNotEqual(mixedObject.anyCol, mixedObject3.anyCol)
+        XCTAssertNotEqual(mixedObject.anyCol.dictionaryValue?["key0"], mixedObject3.anyCol.dictionaryValue?["key0"])
+        XCTAssertNotEqual(mixedObject.anyCol.dictionaryValue?["key0"]?.dictionaryValue?["key5"]?.listValue?[0], mixedObject3.anyCol.dictionaryValue?["key0"]?.dictionaryValue?["key5"]?.listValue?[0])
+    }
+
+    func testMixedCollectionObjectNotifications() throws {
+        let subArray3: AnyRealmValue = AnyRealmValue.fromArray([ .int(3) ])
+        let subArray2: AnyRealmValue = AnyRealmValue.fromArray([ subArray3 ])
+        let subDict1: AnyRealmValue = AnyRealmValue.fromDictionary([ "key1": subArray2 ])
+        let dictionary: Dictionary<String, AnyRealmValue> = [
+            "key0": subDict1,
+        ]
+
+        func expectChange(_ name: String) -> ((ObjectChange<ObjectBase>) -> Void) {
+            let exp = expectation(description: "Object changes for mixed collections")
+            return { change in
+                if case .change(_, let properties) = change {
+                    XCTAssertEqual(properties.count, 1)
+                    if let prop = properties.first {
+                        XCTAssertEqual(prop.name, name)
+                    }
+                } else {
+                    XCTFail("expected .change, got \(change)")
+                }
+                exp.fulfill()
+            }
+        }
+
+        func assertObjectNotification(_ object: Object, block: @escaping () -> Void) {
+            let token = object.observe(expectChange("anyValue"))
+            let realm = realmWithTestPath()
+            try! realm.write {
+                block()
+            }
+
+            waitForExpectations(timeout: 2)
+            token.invalidate()
+        }
+
+        let o = AnyRealmTypeObject()
+        let realm = realmWithTestPath()
+        try realm.write {
+            realm.add(o)
+        }
+
+        assertObjectNotification(o, block: {
```

Comment:
This is a place where it'd be usual to use trailing closure syntax (i.e. `assertObjectNotification(o) {`).